src/cmd/serial.test.js

```diff
@@ -53,5 +61,21 @@ describe('Serial Command', () => {
 			});
 		});
 	});
+
+	describe('serialWifiConfig', () => {
+		it('can reject with timeout after 5000ms if not getting any serial data', () => {
+			clock = sinon.useFakeTimers();
+			const device = { port: 'baltimore' };
+			const mockSerial = new MockSerial();
+			serial.serialPort = mockSerial;
+			const wifiPromise = serial._serialWifiConfig(device);
+			// This allows _serialWifiConfig's internal promises to run and try to
+			// connect to the serial device before moving time forward.
+			process.nextTick(() => {
+				clock.tick(5010);
+			});
+			return expect(wifiPromise).to.be.rejected;
```

Comment:
I should change this to `rejectedWith()` and provide the error string expected. Felt bad to hardcode it in two spots though. Would love a better suggestion for that.

Reply:
yeah, errors are a bit of a mess at the moment. my aim (as noted above) is to start using _actual_ error types w/ codes and so forth. i think given the state of things what you have is fine. if anything, i'd assert against the string but w/e :+1:

not something actionable but fwiw i try to stay away from `chai` and `sinon` extensions - i've just found them more trouble than they're worth. hopefully soon (🤞😬🤞) we'll be able to use more modern js where i've found this pattern works pretty well:

```js
it('does the bar', async () => {
    let error;

    try {
        await foo.bar();
    } catch (e){
        error = e;
    }

    expect(error).to.be.an.instanceof(Error);
    expect(error).to.have.property('message', 'forgot to foo');
    // etc...
});
```